File: src/talker.cpp
#include <ros/ros.h>
#include "std_msgs/String.h"
#include <sstream>

int main(int argc,char **argv) {
    
    ros::init(argc, argv, "talker");    

    ros::NodeHandle nh ;
    ros::Publisher pub = nh.advertise<std_msgs::String>("chatter",10) ;

    ros::Rate loop_rate(10) ;
    
    int count =0 ;
    while(ros::ok()) {
        std_msgs::String msg ;
        msg.data = "hello world" ;

        ROS_INFO("%s", msg.data.c_str()) ;

        pub.publish(msg) ;

        loop_rate.sleep() ;

     
    }


    return 0 ;
}

File: scripts/talker.py
import rospy
from std_msgs.msg import String

def talker() :
    rospy.init_node("talker",anonymous=True) 
    pub = rospy.Publisher('chatter', String , queue_size=10)

    rate = rospy.Rate(20) 
    rospy.loginfo("i am here")

    while not rospy.is_shutdown() :
        hello_str = "helllo " 
        pub.publish(hello_str)
        rospy.loginfo("sending.....")
        rate.sleep()
    
        
        

if __name__ == '__main__' :
    try :
        talker()
    except rospy.ROSInternalException:
           pass
        
